packages/bpk-component-calendar/src/Week.js

```diff
@@ -80,6 +78,61 @@ function getSelectedDate(date, selectionConfiguration) {
   return false;
 }
 
+/**
+ * Gets the correct selection type for the current date
+ * @param {Date} date the current date of the calendar
+ * @param {Object} selectionConfiguration the current selection configuration
+ * @param {Function} formatDateFull function to format dates
+ * @returns {String} selection type to be passed to the date
+ */
+function getSelectionType(date, selectionConfiguration, formatDateFull) {
```

Comment:
Nested are painful which I wanted to avoid but given they return different values it maybe a little more difficult to simplify, could be I add some additional comments around it to help?

For the unit test will see if I can do that, may have to make this function an export in order to do so, but from running jest and its coverage, it gets covered as part of other rendering but will see what I can do

Reply:
Perhaps consider using if/elseif rather than just if. That way it will be easier to follow through what is part of the same logical block.

Reply:
Sure thing! Looking at the code again, I swear I had done that but for some reason looks like `eslint` is trying to be clever and undo these every time I save, so may need to tweak some settings locally and run the eslint cli to see why its suggesting that and what rule it thinks is right!

Reply:
So trying to change it to an if/elseif `eslint` looks to be trying to be smart and avoid an issue with its rules - see screenshot below of what is thrown when changing to if/elseif

![Screenshot 2021-11-10 at 17 22 13](https://user-images.githubusercontent.com/8831547/141162232-384ac4b7-8d72-44e3-acf3-4621b07f1bfd.png)